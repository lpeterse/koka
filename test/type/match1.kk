type test<a> {
  Test(arg : a)
}

function xtest( x : test<a> ) : _ a {
  val (test,_) = (x,"hi")
  match(test) {
    Test(argx) -> { println("test"); argx }
  }
}