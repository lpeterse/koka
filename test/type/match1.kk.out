.copy : forall<a> (test<a>, arg : ?a) -> test<a>
Test : forall<a> (arg : a) -> test<a>
arg : forall<a> (test : test<a>) -> a
xtest : forall<a> (x : test<a>) -> console a